src/DBConnection2.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "DBConnection.hpp"
+
+class IConnectionDispatcher;
+
+class IDBConnection2 {
+public:
+    virtual ~IDBConnection2() = default;
+
+    virtual void connect(IConnectionDispatcher& connectionDispatcher) = 0;
+    virtual int query() const = 0;
+};
+
+class MySqlDBConnection2 : public IDBConnection2 {
+public:
+    MySqlDBConnection2() = default;
+    explicit MySqlDBConnection2(std::string serverVersion, int protocolVersion);
+
+    void connect(IConnectionDispatcher& connectionDispatcher) override;
```

Comment:
@Morheit , method name "connect" doesn't reflect what is does. I propose to name it "dispatch"

Reply:
@malirod , method name `connect` replaced by `dispatch`